### ISSUE TITLE
train on custom data

### ISSUE THREAD

**Issue by ajajar** [ljhandlwt/xnor-net-tf#1]

**Comment by lychahaha** [comment]
Hi @ajajar , Yes, you can train on your custom image after you implement the interface of `Dataset`, including `reset(), get_batch(), is_end()`.
However, this work does not work stably and it can't reduce the forward time as it doesn't implement conv by XNOR op.

**Comment by ajajar** [comment]
Hi @ljhandlwt understand. Thank you very much.

Also in your init_model function it is specific to your application , it s bit different from the pytorch xnor implementation (alexnet binary part )

Thanks

**Comment by lychahaha** [comment]
@ajajar 
Yes, this is NIN, the same as the implement in cifar-10 of pytorch version, not Alexnet.
I don't have enough resource to run the imagenet program, so i give up the implement of Alexnet.
However, i think it's easy to update it from NIN to Alexnet, and maybe some additional details should be took care in pytorch version.
